### PR TITLE
Update resource-metrics-pipeline.md

### LINKED ISSUE #32183: Clarify Metrics API as Beta
**This is a Feature Request**

**What would you like to be added**

[Metrics API](https://kubernetes.io/docs/tasks/debug-application-cluster/resource-metrics-pipeline/#metrics-api) section describes the Metrics API. It should be designated as `beta`.

This PR can be pushed using one of two possible options:

1. Add a sentence after the second to last paragraph on the page indicating that the Metrics API is beta. 

2. Encapsulate this sentence using the Hugo {{< note >}} shortcode. An example using this short code is shown at the bottom of  the page. 

**Why is this needed**

Beta K8s APIs should be explicitly noted as beta in the K8s docs.

### PR DESCRIPTION
**Pull request by PriyanshuAhlawat**
This pr adds a description that Metrics API is beta.
closes #32183

### CODE REVIEW

File: content/en/docs/tasks/debug-application-cluster/resource-metrics-pipeline.md [hunk 1]
```diff
@@ -77,22 +77,24 @@ The architecture components, from right to left in the figure, consist of the fo
...
   If you use a container runtime that uses another resource isolation mechanism, for example
   virtualization, then that container runtime must support
   [CRI Container Metrics](https://github.com/kubernetes/community/blob/master/contributors/devel/sig-node/cri-container-stats.md)
   in order for metrics to be available to the kubelet.
   {{< /note >}}
-  
+
 <!-- body -->
 
 ## Metrics API
```
Comment (tengqm) [inline content/en/docs/tasks/debug-application-cluster/resource-metrics-pipeline.md:92]: You may want to add something like the following here:

```
{{< feature-state for_k8s_version="1.24" state="beta" >}}
```

Comment (sftim) [reply content/en/docs/tasks/debug-application-cluster/resource-metrics-pipeline.md:92]: I'm not sure which Kubernetes release promoted `metrics.kubernetes.io` to beta - it was a while ago, right?

Comment (tengqm) [reply content/en/docs/tasks/debug-application-cluster/resource-metrics-pipeline.md:93]: Right. Just checked this: https://github.com/kubernetes/metrics/commit/5c7f1617ad5e9ceada708f5be3f93918d7b724a4

It was a loooooong time ago.

**Comment by tengqm** [comment]
/hold
The version of the metrics API is independent from kubernetes release, AFAICT. The `feature-state` shortcode is not the right way to "fix" it. I'm wondering if we need to highlight that the API is beta or not.

**Comment by tengqm** [comment]
/hold cancel
It is maybe not a bad thing to use `feature_state` shortcode.
/approve

File: content/en/docs/tasks/debug-application-cluster/resource-metrics-pipeline.md [hunk 2]
```diff
@@ -77,22 +77,23 @@ The architecture components, from right to left in the figure, consist of the fo
...
+{{< feature-state for_k8s_version="1.24" state="beta" >}}
```
Comment (sftim) [inline content/en/docs/tasks/debug-application-cluster/resource-metrics-pipeline.md:92]: Is this right?
```suggestion
{{< feature-state for_k8s_version="1.8" state="beta" >}}
```

**Comment by sftim** [comment]
/hold
but for a different reason: to make sure we don't merge this from an incautious /lgtm

It's OK to unhold this provided you've read through the history of reviews and are satisfied you have all the context.

File: content/en/docs/tasks/debug-application-cluster/resource-metrics-pipeline.md (hunk 1 above)
Comment (dgrisonnet) [reply content/en/docs/tasks/debug-application-cluster/resource-metrics-pipeline.md:93]: Yes the graduation to GA is long due, we have a KEP for it now at least: https://github.com/kubernetes/enhancements/issues/3048

**Comment by chrismetz09** [comment]
PR reads okay to me.

**Review by logicalhan — APPROVED** [review]
/lgtm
/approve

**Pull request merged**